# Andrei Carvalho Torres Portugal

## Disciplina: Probabilidade e Estatística

### Bibliotecas Utilizadas:
- numpy
- pandas

**QUESTÃO 1 A)**

In [24]:
import numpy as np

# Dados da amostra
dados = np.array([11, 39, 27, 25, 41, 18, 32, 35, 21, 19, 12, 42, 4, 5, 40, 32])

# Cálculo das medidas estatísticas
media = np.mean(dados)
variancia = np.var(dados, ddof=1)  # ddof=1 para amostra
desvio_padrao = np.sqrt(variancia)

media, variancia, desvio_padrao


(np.float64(25.1875), np.float64(164.9625), np.float64(12.843772810198724))

Os cálculos resultaram em:

Média (𝜇̂): 25.19

Variância (𝜎²̂): 164.96

Desvio Padrão (𝜎̂): 12.84


**QUESTÃO 1 B)**

Hipotese nula : a média das mensagens enviadas é : 28

Hipotese alternativa : a média não é 28

In [25]:
from scipy import stats

# Parâmetros do teste
mu_hipotetico = 28  # Média proposta na hipótese nula
n = len(dados)  # Tamanho da amostra
graus_de_liberdade = n - 1  # Graus de liberdade para o teste t

# Estatística t
t_stat = (media - mu_hipotetico) / (desvio_padrao / np.sqrt(n))

# Valor crítico para um teste bicaudal com alfa = 0.05
t_critico = stats.t.ppf(1 - 0.025, graus_de_liberdade)  # 0.025 de cada lado

# p-valor associado ao teste t
p_valor = 2 * (1 - stats.t.cdf(abs(t_stat), graus_de_liberdade))

t_stat, t_critico, p_valor


(np.float64(-0.8759108531619951),
 np.float64(2.131449545559323),
 np.float64(0.39488908833634007))

Os resultados do teste são:

Estatística 𝑡: 0.876

Valor critico : ±2.131

p-valor : 0.395

Conclusão : Como "Estatistica t" = 0.876 não ultrapassa o valor critico de 2.131, e o "p-valor" é maior que 0.05, não rejeitamos a "Hipótese Nula".
Ou seja, não há evidências suficientes para afirmar que a média real da rede A seja diferente de 28.

**QUESTÃO 2 A) Cálculo da Média**

In [26]:
# Dados da tabela
fi = np.array([55, 73, 128, 73, 36])  # Frequências
xi = np.array([11, 13, 15, 17, 19])   # Pontos médios

# Cálculo da média
media_tabela = np.sum(xi * fi) / np.sum(fi)
media_tabela


np.float64(14.791780821917808)

Média de atendimentos diários é aproximadamente 14.79

**QUESTÃO 2 B) Cálculo da Variância**

In [27]:
# Dados adicionais da tabela
xi_quadrado_fi = np.array([6655, 12337, 28800, 21097, 12996])  # xi² * fi

# Cálculo da variância
variancia_tabela = (np.sum(xi_quadrado_fi) / np.sum(fi)) - (media_tabela ** 2)
variancia_tabela


np.float64(5.545685869769187)

A variância é aproximadamente 5.55

**QUESTÃO 2 C) Construção do Intervalo de Confiança (95%)**

In [28]:
# Cálculo do desvio padrão
desvio_padrao_tabela = np.sqrt(variancia_tabela)

# Valor crítico t para 95% de confiança e 364 graus de liberdade
t_critico_95 = stats.t.ppf(1 - 0.025, df=364)

# Margem de erro
margem_erro = t_critico_95 * (desvio_padrao_tabela / np.sqrt(365))

# Intervalo de confiança
IC_inf = media_tabela - margem_erro
IC_sup = media_tabela + margem_erro

IC_inf, IC_sup


(np.float64(14.549384671313845), np.float64(15.03417697252177))

O intervalo de confiança (95%) para a média do numero de atendimentos diários é : ( 14.55, 15.03)

**QUESTÃO 2 D) Teste de Hipótese**

Hipotese nula : A média do número de atendimentos é um valor específico ( supondo que seja 15)

Hipótese Alternativa: A média é diferente de 15

In [29]:
# Hipótese testada: média = 15
mu_teste = 15

# Estatística t para o teste
t_stat_teste = (media_tabela - mu_teste) / (desvio_padrao_tabela / np.sqrt(365))

# p-valor associado ao teste t
p_valor_teste = 2 * (1 - stats.t.cdf(abs(t_stat_teste), df=364))

t_stat_teste, p_valor_teste


(np.float64(-1.6892328840687683), np.float64(0.09203112263234492))

Os resultados do teste:

Estatística t: -1.689

p-valor: 0.092

Conclusão : O "p-valor" é 0.092, ou seja, maior que 0.05. Significa que não rejeitamos a "Hipotese nula" ao nível de 5% de significancia. Portanto, não há evidências estatísticas suficientes para afirmar que a média dos atendimentos diários seja diferente de 15


**QUESTÃO 3 Regressão e Correlação**

In [30]:
# Dados do problema
X = np.array([1, 2, 3, 4, 5, 6])
Y = np.array([9500, 16000, 23000, 38000, 62000, 105000])

# Cálculo dos coeficientes da regressão linear
n = len(X)
soma_X = np.sum(X)
soma_Y = np.sum(Y)
soma_XY = np.sum(X * Y)
soma_X2 = np.sum(X**2)

# Coeficiente angular (b)
b = (n * soma_XY - soma_X * soma_Y) / (n * soma_X2 - soma_X**2)

# Coeficiente linear (a)
a = (soma_Y - b * soma_X) / n

a, b


(np.float64(-20800.0), np.float64(18014.285714285714))

O modelo de regressão linear encontrado é:

Y= -20800 + 18014.29X

O que significa : A cada ano (X) que passa, o numero de transistores aumenta em aproximadamente 18.014 unidades. Ademais, o valor inicial previsto (quando X=0) seria -20.800.


Agora vamos -> calcular o Coeficiente de Correlação

In [31]:
# Cálculo do coeficiente de correlação (r)
soma_Y2 = np.sum(Y**2)

# Fórmula de Pearson
r = (n * soma_XY - soma_X * soma_Y) / np.sqrt((n * soma_X2 - soma_X**2) * (n * soma_Y2 - soma_Y**2))

r


np.float64(0.9363102747172248)

Coeficiente de correlação encontrado é : 0.936

Como o Coeficiente de correlação está proximo de 1, há uma forte correlação positiva entre o número de transistores e o tempo. Podemos afirmar que o modelo de regressão linear é adequado para prever o crescimento do número de transistores ao longo dos anos.

**QUESTÃO 4 Regressão e Correlação**

In [32]:
import numpy as np
import pandas as pd

# Dados fornecidos
X = np.array([2, 4, 6, 8, 10, 12, 14, 16, 18, 20])
Y = np.array([20, 28, 35, 48, 54, 58, 60, 61, 60, 62])

# Número de observações
n = len(X)

# Cálculo das somas necessárias
sum_X = np.sum(X)
sum_Y = np.sum(Y)
sum_XY = np.sum(X * Y)
sum_X2 = np.sum(X**2)
sum_Y2 = np.sum(Y**2)

# Coeficiente angular (b)
b = (n * sum_XY - sum_X * sum_Y) / (n * sum_X2 - sum_X**2)

# Coeficiente linear (a)
a = (sum_Y - b * sum_X) / n

# Equação da reta
equation = f"Y = {a:.2f} + {b:.2f}X"

# Coeficiente de correlação de Pearson (r)
r = (n * sum_XY - sum_X * sum_Y) / np.sqrt((n * sum_X2 - sum_X**2) * (n * sum_Y2 - sum_Y**2))

# Estimativa de Y para X = 24
X_pred = 24
Y_pred = a + b * X_pred

# Criar um DataFrame para mostrar os resultados
df_results = pd.DataFrame({
    "Coeficiente Angular (b)": [b],
    "Coeficiente Linear (a)": [a],
    "Equação da Regressão": [equation],
    "Coeficiente de Correlação (r)": [r],
    "Estimativa para X=24": [Y_pred]
})

# Exibir os resultados
print(df_results)

   Coeficiente Angular (b)  Coeficiente Linear (a) Equação da Regressão  \
0                 2.339394               22.866667    Y = 22.87 + 2.34X   

   Coeficiente de Correlação (r)  Estimativa para X=24  
0                       0.919001             79.012121  


**QUESTÃO 5 Distribuição de Poisson**

In [33]:
import numpy as np
import pandas as pd
# Definir a taxa lambda

lambda_ = 0.5  # mensagens por minuto

# Calcular P(T > 2) = e^(-lambda * 2)
P_T_maior_2 = np.exp(-lambda_ * 2)

# Calcular P(T < 4) = 1 - e^(-lambda * 4)
P_T_menor_4 = 1 - np.exp(-lambda_ * 4)

# Criar um DataFrame para mostrar os resultados
df_poisson = pd.DataFrame({
    "Probabilidade de T > 2 minutos": [P_T_maior_2],
    "Probabilidade de T < 4 minutos": [P_T_menor_4]
})

# Exibir os resultados
print(df_poisson)


   Probabilidade de T > 2 minutos  Probabilidade de T < 4 minutos
0                        0.367879                        0.864665


a) maior que 2 minutos é de aproximadamente 36,79%

b)probabilidade de o intervalo ser menor que 4 minutos é de cerca de 86,47%

**QUESTÃO 06 – Distribuição Normal**

In [34]:
from scipy.stats import norm

# Definir os parâmetros
mu = 50  # Média
sigma = 25  # Desvio padrão
X = 40  # Valor de interesse

# Calcular o escore Z
Z = (X - mu) / sigma

# Encontrar a probabilidade acumulada P(X < 40)
P_X_menor_40 = norm.cdf(Z)  # cdf fornece a probabilidade acumulada

# Criar um DataFrame para mostrar os resultados
df_normal = pd.DataFrame({
    "Escore Z": [Z],
    "Probabilidade de X < 40 minutos": [P_X_menor_40]
})

# Exibir os resultados
print(df_normal)


   Escore Z  Probabilidade de X < 40 minutos
0      -0.4                         0.344578


a probabilidade de um candidato terminar o teste em menos de 40 minutos é aproximadamente 34,46%. Isso significa que cerca de 1 em cada 3 candidatos consegue concluir o exame antes desse tempo.

**QUESTÃO 7 - Distribuição Binomial**

In [35]:
# Definir os parâmetros
n = 9  # Número de computadores na amostra
p = 0.10  # Probabilidade de um computador apresentar defeito

# Calcular P(0 defeituosos) = (1 - p)^n
P_0 = (1 - p) ** n

# Calcular P(ao menos 1 defeituoso) = 1 - P(0)
P_1_ou_mais = 1 - P_0

# Criar um DataFrame para mostrar os resultados
df_binomial = pd.DataFrame({
    "Probabilidade de Nenhum Defeituoso (P(0))": [P_0],
    "Probabilidade de Pelo Menos 1 Defeituoso (P(≥1))": [P_1_ou_mais]
})

# Exibir os resultados
print(df_binomial)


   Probabilidade de Nenhum Defeituoso (P(0))  \
0                                    0.38742   

   Probabilidade de Pelo Menos 1 Defeituoso (P(≥1))  
0                                           0.61258  


a probabilidade de que ao menos um dos 9 computadores tenha defeito é de aproximadamente 61,26%. Isso significa que, ao coletar uma amostra de 9 computadores, há uma alta chance de que pelo menos um deles esteja defeituoso.

**QUESTÃO 8 - Distribuição Exponencial**

In [36]:
# Definir os parâmetros
lambda_ = 1 / 4  # Taxa da distribuição exponencial (1/média)
t = 4  # Tempo específico

# Calcular a função de densidade de probabilidade (PDF) para t = 4
pdf_t_4 = lambda_ * np.exp(-lambda_ * t)

# Criar um DataFrame para mostrar os resultados
df_exponencial = pd.DataFrame({
    "Taxa Lambda (1/média)": [lambda_],
    "PDF para T = 4 minutos": [pdf_t_4],
    "Probabilidade P(T=4)": ["Sempre Zero"]
})

# Exibir os resultados
print(df_exponencial)

   Taxa Lambda (1/média)  PDF para T = 4 minutos Probabilidade P(T=4)
0                   0.25                 0.09197          Sempre Zero


 probabilidade de um tempo exato ser observado é sempre zero, pois trata-se de uma variável contínua. No entanto, a função de densidade de probabilidade (PDF) para 𝑡=4 é aproximadamente 0,09197, o que indica a "altura" da distribuição nesse ponto.

**QUESTÃO 9 - Distribuição Uniforme**

In [37]:
# Definir os limites do intervalo
a = 2  # Menor comprimento possível
b = 8  # Maior comprimento possível

# Calcular a média da distribuição uniforme
media_uniforme = (a + b) / 2

# Criar um DataFrame para mostrar os resultados
df_uniforme = pd.DataFrame({
    "Menor Comprimento (a)": [a],
    "Maior Comprimento (b)": [b],
    "Comprimento Médio (E(X))": [media_uniforme]
})

# Exibir os resultados
print(df_uniforme)


   Menor Comprimento (a)  Maior Comprimento (b)  Comprimento Médio (E(X))
0                      2                      8                       5.0


O comprimento médio esperado de um cabo é 5 metros. Isso ocorre porque, em uma distribuição uniforme, a média é simplesmente o ponto médio entre o menor e o maior valor possível.